.github/workflows/end2end.yaml

```diff
@@ -187,9 +187,25 @@ jobs:
           password: ${{ secrets.ARTIFACTS_PASSWORD }}
           source: ${{ env.ROOT_DOCUMENTATION_BUILDDIR }}
 
+
+  mongo-patches-check:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: run patches
+        run: make -C solution-base/mongodb patch
+      - name: Ensure mongodb charts are patched
+        run: |
+          git diff --exit-code || { echo "Error: Changes detected after running 'make fetch-mongodb-sharded'. Please ensure step is completed."; exit 1; }
+        if: always()
+
   build-iso:
     runs-on: ubuntu-22.04
-    needs: [build-kafka, check-dashboard-versions]
+    needs:
+      - build-kafka
+      - check-dashboard-versions
+      - mongo-patches-check
```

Comment:
`mongo-patches-check` should not block `build-iso`, there is no dependency

Reply:
IMHO it's better to have it as a condition; if the patches are not applied, we should block any ISO generation, as the resulting base iso will not work as expected - and the whole build would fail anyway. It's not slowing down the build either.